jq255.md

```diff
@@ -0,0 +1,1425 @@
+specified here, it is possible to produce such `Q`, `sig`, `M0` and `M1`
+with effort about `2^64` operations (i.e. a substantial but not
+infeasible computation). Formally speaking, the scheme achieves normal
+unforgeability but not semi-strong unforgeability. This does *not*
+contradict non-repudiation: in such a scenario, the signer truly signed
+both messages `M0` and `M1`, and can truthfully be held accountable to
+both. However, this lack of message unicity can complicate and even
+formally weaken some protocols such as distributed threshold signature
+schemes with some potentially actively dishonest signers. For "normal"
+usages of signatures (e.g. for authentication or certification
+purposes), this is a non-issue.
+
+### Key Exchange
+
+A *Diffie-Hellman key exchange* is defined here. Two parties (party 1
+and party 2) generate private/public key pairs (sk1 and Q1 for party 1,
+sk2 and Q2 for party 2). Each party receives a copy of the public key of
+the other party, and combines it with its own private key to produce a
+secret key (of size 32 bytes). If the public keys were computed
+correctly and not altered in transit, then both parties compute the same
+secret key, and that key is not known to outsiders. It can then be used
+for further cryptographic operations, typically symmetric encryption.
+
+**Warning:** Two important caveats apply:
+
+  - The key exchange described here does not, by itself, ensure
+    authentication. An active attacker in position to intercept messages
+    exchanged between the two parties may replace public keys on the
+    wire with values generated by the attacker, and establish distinct
+    secret keys, but both known to the attacker, with each party. The
+    attacker may then decrypt and re-encrypt subsequent messages as they
+    flow. A Diffie-Hellman key exchange may provide security against
+    active attackers only if integrated into an outer protocol that
+    ensures proper authentication (e.g. SSL/TLS uses signatures and
+    certificates on top of a Diffie-Hellman key exchange).
+
+  - While private and public keys used here have the same format as
+    private and public keys used for signatures, it is **not
+    recommended** to use the same private/public key pair for both key
+    exchange and signatures. In general, the two activities call for key
+    pairs with distinct, incompatible lifecycle properties (simply put,
+    keys used for encryption should normally be backed up to prevent
+    data loss in case of key loss, while keys used for signatures must
+    not be backed up, since their value relies in their exclusive
+    control by the purported signer). Moreover, possible interactions
+    between the two schemes are not entirely explored.
+
+The process below describes how a party uses its own key pair (`sk` and
+`Q`) along with the *encoded* public key received from the peer
+(`peer_pk`) to compute the shared secret. The function returns two
```

Comment:
```suggestion
(`peer_PK`) to compute the shared secret. The function returns two
```

Reply:
This `peer_pk` is the name of the parameter to the function in the pseudocode. Are you suggesting it making it uppercase in the pseudocode as well?

Reply:
Yes, if it's a group element/point, uppercase; if it's a scalar/field element, lowercase

Reply:
`PEER_PK` is shouty but

Reply:
Well technically it is an _encoded_ group element (or purported to be, at least) so maybe I should call it `peer_encoded_pk`.

Reply:
Well I have renamed it to `peer_encoded_Q` (the decoded point was already `peer_Q`).